client/app/domain/general-informations/domain-general-informations.controller.js

```diff
@@ -310,6 +314,31 @@ export default class DomainTabGeneralInformationsCtrl {
       });
   }
 
+  getAllOptionDetails(serviceName) {
+    this.loading.options = true;
+    return this.Domain.getOptions(serviceName)
+      .then(options => this.$q.all(
+        _.map(options, option => this.Domain.getOption(serviceName, option)
+          .then(optionDetail => Object.assign({}, optionDetail,
+            { optionActivated: optionDetail.state === this.DOMAIN.DOMAIN_OPTION_STATUS.ACTIVE }))),
+      ))
+      .then((options) => {
+        const transformedOptions = {};
+        _.forEach(options, (option) => {
+          transformedOptions[option.option] = option;
```

Comment:
Maybe a `_.reduce` would help here 
```suggestion
this.options =  _.reduce(options, (acc, option) => ({
  ...acc,
  [option.option] = option
 })
```

Reply:
Thanks! This is done